### PR TITLE
Maeste/issue331

### PR DESCRIPTION
**Pull request by maeste**

### CODE REVIEW

File: examples/perceive_ask_and_act/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+### Step 4: Experiment
+
+You can experiment with different agent configurations or modify the agent behaviors by editing the YAML files located in the `examples/perceive_ask_and_act` directory. These configuration files define the agents' properties, including memory, communication methods, and response patterns. You can do that without restarting everything, just edit the yaml file(s) and use CLI's /reload command (see our documentation for more details on how to use CLI)
+
+## Understanding the YAML Configuration and custom_agents.py
+
+- **intelligent-agents.yaml**: Defines all the agents discussed in this document and needed to run the example
+- **custom_agents.py**: contain 3 custom agents efine for the purpose of this example. It has also a `main()` function which instantiate the `UserQuestionAgent`
```
Comment (fjuma) [inline examples/perceive_ask_and_act/README.md:66]: s/efine/defined

File: src/wiseagents/agents/coordinator_wise_agents.py [hunk 2]
```diff
@@ -101,6 +101,31 @@ def set_response_delivery(self, response_delivery: Callable[[], WiseAgentMessage
...
+            request (WiseAgentMessage): the request message to process
+        """
+        logging.debug(f"Sequential coordinator received request: {request}")
+
+        # Generate a chat ID that will be used to collaborate on this query
+        chat_id = str(uuid.uuid4())
+
+        ctx = WiseAgentRegistry.get_or_create_context(request.context_name)
+        ctx.set_collaboration_type(chat_id, WiseAgentCollaborationType.SEQUENTIAL_MEMORY)
+        ctx.append_chat_completion(chat_uuid=chat_id, messages={"role": "system", "content": "hello"})
```
Comment (fjuma) [inline src/wiseagents/agents/coordinator_wise_agents.py:124]: Just to check, do we need this hardcoded hello message?

**Review by fjuma — CHANGES_REQUESTED** [review]

File: src/wiseagents/agents/coordinator_wise_agents.py (hunk 2 above)
Comment (maeste) [reply src/wiseagents/agents/coordinator_wise_agents.py:124]: We need a system message, but it's better to use the one declared for agent or llm
ctx.append_chat_completion(chat_uuid=chat_id, messages={"role": "system", "content": self.system_message or self.llm.system_message})
Fixing

File: src/wiseagents/agents/coordinator_wise_agents.py [hunk 3]
```diff
@@ -101,6 +101,32 @@ def set_response_delivery(self, response_delivery: Callable[[], WiseAgentMessage
...
+            request (WiseAgentMessage): the request message to process
+        """
+        logging.debug(f"Sequential coordinator received request: {request}")
+
+        # Generate a chat ID that will be used to collaborate on this query
+        chat_id = str(uuid.uuid4())
+
+        ctx = WiseAgentRegistry.get_or_create_context(request.context_name)
+        ctx.set_collaboration_type(chat_id, WiseAgentCollaborationType.SEQUENTIAL_MEMORY)
+        ctx.append_chat_completion(chat_uuid=chat_id, messages={"role": "system", "content": self.system_message or self.llm.system_message})
```
Comment (fjuma) [inline src/wiseagents/agents/coordinator_wise_agents.py:124]: system_message should be added to the `__init__` method as an `Optional[str]` and it should also be added in the `__new__` method

File: src/wiseagents/agents/coordinator_wise_agents.py [hunk 4]
```diff
@@ -101,6 +101,44 @@ def set_response_delivery(self, response_delivery: Callable[[], WiseAgentMessage
...
+        obj = super().__new__(cls)
+        obj._system_message = None
+        return obj
+
+    def __init__(self, name: str, description: str, transport: WiseAgentTransport, agents: List[str], system_message: Optional[str] = None):
+        self._system_message = system_message
+        super().__init__(name, description, transport, agents)
+
+    @property
+    def system_message(self) -> str:
```
Comment (fjuma) [inline src/wiseagents/agents/coordinator_wise_agents.py:118]: Last small comment, we can remove this method and pass the system message in the super().__init__ call above.

**Review by fjuma — APPROVED** [review]

**Pull request merged**